ts/test/session/sending/MessageSender_test.ts

```diff
@@ -0,0 +1,176 @@
+import { expect } from 'chai';
+import * as crypto from 'crypto';
+import * as sinon from 'sinon';
+import { toNumber } from 'lodash';
+import { MessageSender } from '../../../session/sending';
+import LokiMessageAPI from '../../../../js/modules/loki_message_api';
+import { TestUtils } from '../../test-utils';
+import { UserUtil } from '../../../util';
+import { MessageEncrypter } from '../../../session/crypto';
+import { SignalService } from '../../../protobuf';
+import LokiPublicChatFactoryAPI from '../../../../js/modules/loki_public_chat_api';
+import { OpenGroupMessage } from '../../../session/messages/outgoing';
+import { LokiPublicChannelAPI } from '../../../../js/modules/loki_app_dot_net_api';
+
+describe('MessageSender', () => {
+  const sandbox = sinon.createSandbox();
+
+  afterEach(() => {
+    sandbox.restore();
+    TestUtils.restoreStubs();
+  });
+
+  describe('send', () => {
+    const ourNumber = 'ourNumber';
+    let lokiMessageAPIStub: sinon.SinonStubbedInstance<LokiMessageAPI>;
+    let stubEnvelopeType = SignalService.Envelope.Type.CIPHERTEXT;
+
+    beforeEach(() => {
+      // We can do this because LokiMessageAPI has a module export in it
+      lokiMessageAPIStub = sandbox.createStubInstance(LokiMessageAPI, {
+        sendMessage: sandbox.stub(),
+      });
+      TestUtils.stubWindow('lokiMessageAPI', lokiMessageAPIStub);
+
+      sandbox.stub(UserUtil, 'getCurrentDevicePubKey').resolves(ourNumber);
+      sandbox
+        .stub(MessageEncrypter, 'encrypt')
+        .callsFake(async (_device, plainTextBuffer, _type) => ({
+          envelopeType: stubEnvelopeType,
+          cipherText: plainTextBuffer,
+        }));
```

Comment:
why not use SessionCipherStub here?

Reply:
because `MessageEncrypter` and `SessionCipher` aren't the same thing